### PR TITLE
Compass structured error

### PR DESCRIPTION
**Pull request by maycommit**

### CODE REVIEW

File: compass/internal/metricsgroup/metricsgroup.go [hunk 1]
```diff
@@ -20,20 +20,21 @@ package metricsgroup
...
 import (
 	"encoding/json"
-	"errors"
-	"fmt"
-	"github.com/ZupIT/charlescd/compass/internal/metric"
-	"github.com/ZupIT/charlescd/compass/internal/metricsgroupaction"
-	"github.com/ZupIT/charlescd/compass/internal/util"
-	"github.com/ZupIT/charlescd/compass/pkg/datasource"
-	"github.com/ZupIT/charlescd/compass/pkg/logger"
+	// "errors"
```
Comment (gabrielleitezup) [inline compass/internal/metricsgroup/metricsgroup.go:23]: Remove comment

**Review by gabrielleitezup — CHANGES_REQUESTED** [review]

File: compass/internal/metricsgroupaction/actionexecution.go [hunk 2]
```diff
@@ -122,8 +116,8 @@ func (main Main) SetExecutionSuccess(actionExecutionID string, executionLog stri
 	execution.ExecutionLog = executionLog
 	result := main.db.Save(&execution)
 	if result.Error != nil {
-		logger.Error(util.SetExecutionSuccessError, "SetActionExecutionSuccess", err, fmt.Sprintf("%+v", execution))
-		return ActionsExecutions{}, err
+		return ActionsExecutions{}, errors.NewError("Set error", result.Error.Error()).
+			WithOperations("SetExecutionSuccess.validateActionCanFinish")
```
Comment (gabrielleitezup) [inline compass/internal/metricsgroupaction/actionexecution.go:120]:
```suggestion
			WithOperations("SetExecutionSuccess.Save")
```

File: compass/web/api/api.go [hunk 3]
```diff
@@ -1,70 +1,89 @@
...
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- *
- */
-
 package api
```
Comment (gabrielleitezup) [inline compass/web/api/api.go:1]: Copyright removed

File: compass/web/api/middleware.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+	}
+
+	permissions, err := api.mooveMain.GetUserPermissions(user.ID, workspaceID)
+	if err != nil {
+		return false, err.WithOperations("authorizeUser.GetUserPermissions")
+	}
+
+	for _, permission := range permissions {
+		allowed, enforcerErr := api.enforcer.Enforce(permission, url, method)
+		if err != nil {
```
Comment (gabrielleitezup) [inline compass/web/api/middleware.go:146]: The `if` should check the `enforcerErr`, not the `err`

File: compass/web/api/v1.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	}
+	{
+		path := "/metrics-groups"
+		s.HandleFunc(path, metricsgroup.Create(api.metricsGroupMain)).Methods("POST")
+		s.HandleFunc(path, metricsgroup.GetAll(api.metricsGroupMain)).Methods("GET")
+		s.HandleFunc(fmt.Sprintf("%s/{metricGroupID}", path), metricsgroup.Show(api.metricsGroupMain)).Methods("GET", "PUT")
+		s.HandleFunc(fmt.Sprintf("%s/{metricGroupID}/query", path), metricsgroup.Query(api.metricsGroupMain)).Methods("GET")
+		s.HandleFunc(fmt.Sprintf("%s/{metricGroupID}/result", path), metricsgroup.Result(api.metricsGroupMain)).Methods("GET")
+		s.HandleFunc(fmt.Sprintf("%s/{metricGroupID}", path), metricsgroup.Update(api.metricsGroupMain)).Methods("PUT")
+		s.HandleFunc(fmt.Sprintf("%s/{metricGroupID}", path), metricsgroup.UpdateName(api.metricsGroupMain)).Methods("PATCH") // TODO: Discutir necessidade desse patch
```
Comment (gabrielleitezup) [inline compass/web/api/v1.go:50]: Keep TODO?

**Review by gabrielleitezup — CHANGES_REQUESTED** [review]

File: compass/web/api/v1.go (hunk 5 above)
Comment (arthurpuccizup) [reply compass/web/api/v1.go:50]: Lets keep, for later discussion

File: compass/web/api/middleware.go (hunk 4 above)
Comment (arthurpuccizup) [reply compass/web/api/middleware.go]: Done

File: compass/web/api/api.go (hunk 3 above)
Comment (arthurpuccizup) [reply compass/web/api/api.go:19]: Done

File: compass/internal/metricsgroupaction/actionexecution.go (hunk 2 above)
Comment (arthurpuccizup) [reply compass/internal/metricsgroupaction/actionexecution.go]: Done

File: compass/internal/metricsgroup/metricsgroup.go (hunk 1 above)
Comment (arthurpuccizup) [reply compass/internal/metricsgroup/metricsgroup.go]: Done

**Review by gabrielleitezup — APPROVED** [review]

File: compass/internal/health/health.go [hunk 6]
```diff
@@ -90,26 +91,25 @@ func (main Main) getComponentStatus(thresholdValue, metricValue float64) string
 	return "STABLE"
 }
 
-func (main Main) getComponentsErrorPercentage(circleIDHeader, circleId string, workspaceID uuid.UUID) ([]CircleComponentHealthRepresentation, error) {
+func (main Main) getComponentsErrorPercentage(circleIDHeader, circleId string, workspaceID uuid.UUID) ([]CircleComponentHealthRepresentation, errors.Error) {
 	body, err := main.mooveMain.GetMooveComponents(circleIDHeader, circleId, workspaceID)
 	if err != nil {
-		logger.Error(util.QueryGetPluginError, "getComponentsErrorPercentage", err, nil)
-		return nil, err
+		return nil, err.WithOperations("getResultQuery.Lookup")
```
Comment (gabrielleitezup) [inline compass/internal/health/health.go:97]:
```suggestion
		return nil, err.WithOperations("getComponentsErrorPercentage.GetMooveComponents")
```

**Review by gabrielleitezup — CHANGES_REQUESTED** [review]

**Review by gabrielleitezup — APPROVED** [review]

**Review by icaroafonsozup — APPROVED** [review]

**Pull request merged**